### ISSUE TITLE
Configure default severity

### ISSUE THREAD

**Issue by Bobronium** [DavidAnson/vscode-markdownlint#241]
I'm trying to configure default rules severity, but I guess, it doesn't work like that:
```js
    "markdownlint.config": {
        "default": {
            "severity": "Information"
        },
    },
```
Is there a way to do it?

Upd. 

Ok, I see now that severity is hardcoded here:
https://github.com/DavidAnson/vscode-markdownlint/blob/522848a8cfe336bbc58056689e1a09f7dc8564f5/extension.js#L598

Can we make it configurable?

**Comment by DavidAnson** [comment]
Why do you want this? The current use of `Warning` seems appropriate according to the VS Code definitions:

```js
	/**
	 * Represents the severity of diagnostics.
	 */
	export enum DiagnosticSeverity {

		/**
		 * Something not allowed by the rules of a language or other means.
		 */
		Error = 0,

		/**
		 * Something suspicious but allowed.
		 */
		Warning = 1,

		/**
		 * Something to inform about but not a problem.
		 */
		Information = 2,

		/**
		 * Something to hint to a better way of doing it, like proposing
		 * a refactoring.
		 */
		Hint = 3
	}
```

**Comment by DavidAnson** [comment]
Thanks. What changes if you were able to change these from Warning to Information? Are there different policies that kick in or something?

**Comment by Bobronium** [comment]
It allows in turn to configure a visual scheme for one's taste. 

For instance: https://github.com/Bobronium/vscode-pycharm-darcula-theme#linting-highlighting

**Comment by amyq** [comment]
As a technical writer, I'm interested in this too, and can unpack why. The technical writers at GitLab use markdownlint as part of our linting suite. For rules that we enable, violations cause the `docs-lint markdown` job to fail in CI/CD. Thus, for us any Markdownlint findings are errors and not warnings. 

The **Problems** output in VS Code sorts by severity, so it would be nice to surface these as we do any other error.

We do have a pre-push hook set up that fails when Markdownlint findings are detected, but it would be nice if the findings were flagged as errors in the editor.

**Comment by archifont** [comment]
Hello! First, thank you for this excellent extension – it's been invaluable for me maintaining consistent Markdown quality (trying to write a PhD in Markdown, using pandoc to disseminate, very fun way to procrastinate trying to make typographically nice documents).  

I'd like to +1 this question, for example the way it is similar to how the [Spell Right extension](https://github.com/bartosz-antosik/vscode-spellright)  implements its `spellright.notificationClass` setting.

### Reasons

For users working extensively with Markdown (especially in academic or technical writing contexts), having all violations displayed as warnings can be visually overwhelming. Being able to set violations to `"information"` or `"hint"` level would:

- Reduce visual noise while still providing linting feedback
- Allow users to focus on more critical issues (if using multiple linters)
- Provide a less intrusive linting experience for those who prefer subtler feedback (by the way `focusMode` is already extremely helpful, thanks for that!)

### Current Behavior

Currently, all markdownlint rule violations are displayed as **Warning** diagnostics in VS Code, regardless of rule configuration.

### Proposed Feature (somewhat like the previous comment)

Add a configuration setting (e.g., `markdownlint.diagnosticSeverity`) that would allow users to choose the diagnostic class for violations. This could support VS Code's diagnostic severity levels:

- `"error"` - Error diagnostic
- `"warning"` - Warning diagnostic - **current default**
- `"information"` - Information diagnostic
- `"hint"` - Hint diagnostic (minimal visual distraction)

### Example Configuration

Similar to Spell Right's implementation:

```json
{
  "markdownlint.diagnosticSeverity": "information"
}
```

### Additional Consideration

For even more flexibility (though perhaps as a future enhancement), per-rule severity configuration could be useful:

```json
{
  "markdownlint.config": {
    "MD013": {
      "line_length": 120,
      "severity": "hint"
    },
    "MD041": "information"
    // or whatever syntax is congruent
  }
}
```

I notice that the core markdownlint library added `severity` support in v0.39.0, so this might provide a good foundation for extending these levels, eventually trickling down into VS Code extension?

Thank you for (re)considering this question!
